src/OpenSage.Mods.Generals/Gui/GameOptionsUtil.cs

```diff
@@ -0,0 +1,287 @@
+            _window.Controls.FindControl(_optionsPath + ":MapWindow").Hide();
+            _window.Controls.FindControl(_optionsPath + ":TextEntryMapDisplay").Hide();
+            _window.Controls.FindControl(_optionsPath + ":StaticTextMapPreview").Hide();
+            _window.Controls.FindControl(_optionsPath + ":StaticTextTeam").Hide();
+            _window.Controls.FindControl(_optionsPath + ":StaticTextFaction").Hide();
+            _window.Controls.FindControl(_optionsPath + ":StaticTextColor").Hide();
+
+            for (int i = 0; i < 8; ++i)
+            {
+                _window.Controls.FindControl(_optionsPath + ":ComboBoxTeam" + i.ToString()).Hide();
+                _window.Controls.FindControl(_optionsPath + ":ComboBoxPlayerTemplate" + i.ToString()).Hide();
+                _window.Controls.FindControl(_optionsPath + ":ComboBoxColor" + i.ToString()).Hide();
+            }
+
+            context.WindowManager.PushWindow(@"Menus\" + _mapSelectPath);
+        }
+
+        public void CloseMapSelection(ControlCallbackContext context)
+        {
+            // Reshow controls
+            _window.Controls.FindControl(_optionsPath + ":ButtonSelectMap").Show();
+            _window.Controls.FindControl(_optionsPath + ":MapWindow").Show();
+            _window.Controls.FindControl(_optionsPath + ":TextEntryMapDisplay").Show();
+            _window.Controls.FindControl(_optionsPath + ":StaticTextMapPreview").Show();
+            _window.Controls.FindControl(_optionsPath + ":StaticTextTeam").Show();
+            _window.Controls.FindControl(_optionsPath + ":StaticTextFaction").Show();
+            _window.Controls.FindControl(_optionsPath + ":StaticTextColor").Show();
+
+            for (int i = 0; i < 8; ++i)
+            {
+                _window.Controls.FindControl(_optionsPath + ":ComboBoxTeam" + i.ToString()).Show();
+                _window.Controls.FindControl(_optionsPath + ":ComboBoxPlayerTemplate" + i.ToString()).Show();
+                _window.Controls.FindControl(_optionsPath + ":ComboBoxColor" + i.ToString()).Show();
+            }
+
+            context.WindowManager.PopWindow();
+        }
+
+        private void FillComboBoxOptions(string key, string[] options, int selectedIndex = 0)
+        {
+            var comboBoxs = Control.GetSelfAndDescendants(_window).OfType<ComboBox>().Where(i => i.Name.StartsWith(key));
+            foreach (var comboBox in comboBoxs)
+            {
+                if (comboBox.Name.Length - 1 != key.Length) continue;
+                ListBoxDataItem[] items = options.Select(i =>
+                    new ListBoxDataItem(comboBox, new[] { _game.ContentManager.TranslationManager.Lookup(i) })).ToArray();
+                comboBox.Items = items;
+                comboBox.SelectedIndex = selectedIndex;
+            }
+        }
+
+        private void FillColorComboBoxOptions(KeyValuePair<string, ColorRgbaF>[] options, int selectedIndex = 0)
```

Comment:
Minor thing: Use a tuple instead of `KeyValuePair`.

Reply:
Copied that from @tgjoned 😄 But will do